### PR TITLE
Preliminary HP Virtual Connect support

### PR DESCRIPTION
**Pull request by mieleton**
Barebones support to discovering HP Virtual Connect (Ethernet) modules including their model, serial and version.

SNMP support in Virtual Connect is quite limited, information like CPU and MEM usage etc. is not available.

### CODE REVIEW

**Comment by laf** [comment]
You need to sign the contributors agreement: http://docs.librenms.org/General/Contributing/

**Comment by mieleton** [comment]
@laf done, see pull request https://github.com/librenms/librenms/pull/3489

File: includes/polling/os/hpvc.inc.php [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+// Format of sysDescr is hardware model followed by version followed by build date
+$tempstr = substr($poll_device['sysDescr'], 0, strrpos($poll_device['sysDescr'], ' '));
+$version = trim(substr($tempstr, strrpos($tempstr, ' ')));
+$hardware = trim(substr($tempstr, 0, strrpos($tempstr, ' ')));
+
+// Serial number is in sysName after string "VCEX"
+$serial = substr(snmp_get($device, 'sysName.0', '-OvQ', 'SNMPv2-MIB:HOST-RESOURCES-MIB:SNMP-FRAMEWORK-MIB'), 4);
```
Comment (laf) [inline includes/polling/os/hpvc.inc.php]: sysName should be available in $poll_device['sysName']

File: includes/definitions.inc.php [hunk 2]
```diff
@@ -1577,6 +1577,14 @@
 $config['os'][$os]['over'][0]['graph'] = 'device_bits';
 $config['os'][$os]['over'][0]['text']  = 'Traffic';
 
+// HP Virtual Connect
+$os = 'hpvc';
```
Comment (laf) [inline includes/definitions.inc.php]: Is hpvc the OS name for this?

Comment (mieleton) [inline includes/definitions.inc.php]: HP does not have any (public) OS name for the software the modules are running, so I went with an acronym of HP Virtual Connect to keep it short.

File: includes/polling/os/hpvc.inc.php (hunk 1 above)
Comment (mieleton) [inline includes/polling/os/hpvc.inc.php]: Yes, though looking at the code it gets transformed to lowercase from the original string. I wanted to keep the serial number in exactly the same format it's visible in other places. However, I can change this, if needed.

**Comment by laf** [comment]
We probably should just expose sysName_orig or something but yes for now this is fine.

**Pull request merged**